src/compas/geometry/transformations/quaternions.py

```diff
@@ -0,0 +1,324 @@
+from compas.geometry.transformations import quaternion_from_matrix
+from compas.geometry.transformations import matrix_from_quaternion
+from compas.geometry.transformations import euler_angles_from_matrix
+from compas.geometry.transformations import matrix_from_euler_angles
+from compas.geometry.transformations import axis_and_angle_from_matrix
+from compas.geometry.transformations import matrix_from_axis_and_angle
+
+#from compas.geometry.basic import allclose
+import math
+
+
+__all__ = [
+    'quaternion_norm',
+    'quaternion_unitize',
+    'quaternion_is_unit',
+    'quaternion_multiply',
+    'quaternion_canonic',
+    'quaternion_conjugate',
+    'quaternion_from_euler_angles',
+    'euler_angles_from_quaternion',
+    'quaternion_from_axis_angle',
+    'axis_angle_from_quaternion']
+
+
+"""
+This module contains functions that operate on and/or return quaternions.
+
+Default convention to represent a quaternion in this module is a tuple of four real values [qw,qx,qy,qz].
+The first value qw is the scalar (real) part, and qx,qy,qz form the vector (complex, imaginary) part, so that:
+q = qw + qx*i + qy*j + qz*k,
+where i,j,k are basis components.
+
+Quaternion algebra basics:
+i*i = j*j = k*k = i*j*k = -1
```

Comment:
yes, the format is latex. check the numerical package for examples...

Reply:
where will this module docstring and other comments in the methods in this module apprear in the documentation? it seems the modules in `geometry.transformations` are hidden for now, right?

Reply:
all modules beyond the second module are "hidden" because functionality is meant to be pulled up and used from that second level. otherwise the user needs to be aware of the underlying structure of those modules further down. it also provides the opportunity to move things around without requiring users to change their code.

this means you should add the module level information to the docstring of `compas.geometry.__init__.py`, for example in a section "Quaternions".

it is true that the geometry module is getting quite big and perhaps a bit unmanageable in that sense. however, for now, we will stick to this convention...

Reply:
ok. now the functions appear in the docs, but not the top module docstring, so I copied it into the Quaternion class dosctring, ok?
Still I wonder where will `@property` and operator functions be visible in the documentation?